ged4py/parser.py

```diff
@@ -70,19 +70,25 @@ def guess_codec(file, errors="strict", require_char=False):
     :param str errors: Controls error handling behavior during string
         decoding, accepts same values as standard `codecs.decode` method.
     :param bool require_char: If True then exception is thrown if CHAR
-        record is not found in a header, if False and C?HAR is not in the
-        header then codec determined from BOM or "ansel" is returned.
+        record is not found in a header, if False and CHAR is not in the
+        header then codec determined from BOM or "gedcom" is returned.
     :returns: Tuple (codec_name, bom_size)
     :raises: :py:class:`CodecError` when codec name in file is unknown or
         when codec name in file contradicts codec determined from BOM.
     :raises: :py:class:`UnicodeDecodeError` when codec fails to decode
         input lines and `errors` is set to "strict" (default).
     """
 
+    # mapping of gedcom character set specifiers to Python encoding names
+    gedcom_char_to_codec = {
+        'ansel': 'gedcom',
+        'unicode': 'utf-16',
```

Comment:
This comes from the GEDCOM 5.5.1 draft specification (http://phpgedview.sourceforge.net/ged551-5.pdf). Chapter 3 covers character sets, and calls out UNICODE as one of the supported `CHAR` options. Based on the references to ISO-10646 and a 16-bit encoding, this likely referred to UCS-2 (given the time of publication), which as been superseded by UTF-16.

I'm not aware of any applications that generate GEDCOM files with the UNICODE encoding. I added this entry based on the existing (commented out) tests that implied there was interest in supporting UTF-16 (I can pull it out of this change if you would prefer.

Reply:
It looks indeed that UNICODE section of the draft is from very early UNICODE era and it does not reflect current state of UNICODE. I think I prefer to avoid adding support for this option to `ged4py` unless there is an explicit request for it (e.g. some application which generates that). If you can just drop that entry from the map I think it would be OK.

UTF-16 support is potentially very messy business, I remember that I was initially thinking about adding it but that opened very big can of worms (mostly because GEDCOM specification is vague in many areas) so I don't really want to go there anymore. And it looks like world has converged on UTF-8 anyways, I doubt anyone out there cares much about UTF-16.